lib/graphql/schema/printer.rb

```diff
@@ -44,6 +48,10 @@ def print_schema_definition(schema)
       BUILTIN_SCALARS = Set.new(["String", "Boolean", "Int", "Float", "ID"])
       private_constant :BUILTIN_SCALARS
 
+      def is_spec_directive(directive)
+        ['skip', 'include', 'deprecated'].include?(directive.name)
+      end
```

Comment:
Don't worry, I kept confusing myself here too.

There's two concepts: `Directive` and `DirectiveDefinition`.

Example `@skip` `Directive`:

```
query myQuery($someTest: Boolean) {
  experimentalField @skip(if: $someTest)
}
```

Example `@skip` `DirectiveDefinition`:

```
directive @skip(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT
```

The `is_spec_directive` method is used to omit the `DirectiveDefinition` of spec directives from getting printed when dumping the schema.

I need to rename `def print_directive(directive)` to `def print_directive_definition(directive)` as that's not named correctly and could be a source of confusion.

In terms of `Directive` we're only printing the `deprecated` directive at the moment, see `DeprecatedPrinter`. We may want to generalize this in the future though.